commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java

```diff
@@ -0,0 +1,187 @@
+            {2520.35, 2520.35},
+            {2010.87, 2520.35},
+            {100000000.1, 100000000.1},
+            {+0.0, 100000000.1},
+            {-0.0, 100000000.1},
+            {Double.MAX_VALUE, Double.MAX_VALUE},
+            {Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY},
+            {Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY}
+        };
+
+        Max stat = Max.create();
+        for (final double[] valueAndExpected: testArray) {
+            final double value = valueAndExpected[0];
+            final double expected = valueAndExpected[1];
+            stat.accept(value);
+            Assertions.assertEquals(expected, stat.getAsDouble());
+        }
+    }
+
+    @Test
+    void testNaN() {
+        // Test non-nan values cannot revert a NaN
+        double[] testArray = {Double.NaN, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY};
+        Max stat = Max.create();
+        for (final double x : testArray) {
+            stat.accept(x);
+            Assertions.assertEquals(Double.NaN, stat.getAsDouble());
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void testMax(double[] values, double expected) {
+        Max stat = Max.create();
+        Arrays.stream(values).forEach(stat);
+        double actual = stat.getAsDouble();
+        Assertions.assertEquals(expected, actual, "max");
+        Assertions.assertEquals(expected, Max.of(values).getAsDouble(), "max");
+    }
+
+    static Stream<Arguments> testMax() {
+        return Stream.of(
+            Arguments.of(new double[] {}, Double.NEGATIVE_INFINITY),
+            Arguments.of(new double[] {3.14}, 3.14),
+            Arguments.of(new double[] {12.34, 56.78, -2.0}, 56.78),
+            Arguments.of(new double[] {Double.NaN, 3.14, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {-1d, 1d, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {Double.NaN, Double.NaN, Double.NaN}, Double.NaN),
+            Arguments.of(new double[] {0.0d, Double.NaN, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY}, Double.NaN),
+            Arguments.of(new double[] {+0.0d, -0.0d, 1.0, 3.14}, 3.14),
+            Arguments.of(new double[] {-0.0, +0.0}, 0.0),
+            Arguments.of(new double[] {0.0, -0.0}, 0.0),
+            Arguments.of(new double[] {0.0, +0.0}, 0.0),
+            Arguments.of(new double[] {1.2, -34.56, 456.789, -5678.9012}, 456.789),
+            Arguments.of(new double[] {-23467824, 23648, 2368, 23749, -23424, -23492, -92397747}, 23749),
+            Arguments.of(new double[] {0.0d, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.MIN_VALUE}, Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {0.0d, +0.0d, -0.0d, Double.POSITIVE_INFINITY, -Double.MIN_VALUE}, Double.POSITIVE_INFINITY),
+            Arguments.of(new double[] {0.0d, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY, Double.MIN_VALUE}, Double.POSITIVE_INFINITY)
```

Comment:
Quickly compared the code in this PR with the existing Min code, and everything looks OK. The only part I noticed that did not change was this with the `Double.MIN_VALUE`'s. Is that intentional or were we supposed to use `Double.MAX_VALUE` for this test?

Reply:
Thanks @kinow, I've updated the tests for Max per your feedback.